spec/integration/bisect_spec.rb

```diff
@@ -0,0 +1,158 @@
+          | - Remaining non-failing examples (2):
+          |    - ./spec/rspec/core/resources/order_dependent_specs.rb[10:1,11:1]
+          | - Round finished (n.nnnn seconds)
+          |Round 4: searching for 1 non-failing example (of 2) to ignore:
+          | - Running: rspec ./spec/rspec/core/resources/order_dependent_specs.rb[11:1,22:1] --order defined (n.nnnn seconds)
+          | - Examples we can safely ignore (1):
+          |    - ./spec/rspec/core/resources/order_dependent_specs.rb[10:1]
+          | - Remaining non-failing examples (1):
+          |    - ./spec/rspec/core/resources/order_dependent_specs.rb[11:1]
+          | - Round finished (n.nnnn seconds)
+          |Round 5: searching for 1 non-failing example (of 1) to ignore:
+          | - Running: rspec ./spec/rspec/core/resources/order_dependent_specs.rb[22:1] --order defined (n.nnnn seconds)
+          | - Round finished (n.nnnn seconds)
+          |Bisect complete! Reduced necessary non-failing examples from 21 to 1 in n.nnnn seconds.
+          |
+          |The minimal reproduction command is:
+          |  rspec ./spec/rspec/core/resources/order_dependent_specs.rb[11:1,22:1] --order defined
+        EOS
+      end
+    end
+
+    context "when a load-time problem occurs while running the suite" do
+      it 'surfaces the stdout and stderr output to the user' do
+        output = bisect(%w[spec/rspec/core/resources/fail_on_load_spec.rb_], 1)
+        expect(output).to include("Bisect failed!", "undefined method `contex'", "About to call misspelled method")
+      end
+    end
+
+    context "when the spec ordering is inconsistent" do
+      it 'stops bisecting and surfaces the problem to the user' do
+        output = bisect(%W[spec/rspec/core/resources/inconsistently_ordered_specs.rb], 1)
+        expect(output).to include("Bisect failed!", "The example ordering is inconsistent")
+      end
+    end
+
+    context "when the user aborts the bisect with ctrl-c" do
+      before do
+        formatter_subclass = Class.new(Formatters::BisectProgressFormatter) do
+          Formatters.register self, :bisect_round_finished
+
+          def bisect_round_finished(notification)
+            if notification.round == 2
+              Process.kill("INT", Process.pid)
+              # Process.kill is not a synchronous call, so to ensure the output
+              # below aborts at a deterministic place, we need to block here.
+              # The sleep will be interrupted by the signal once the OS sends it.
+              # For the most part, this is only needed on JRuby, but we saw
+              # the asynchronous behavior on an MRI 2.0 travis build as well.
+              sleep 5
+            else
+              super
+            end
+          end
+        end
+
+        stub_const(Formatters::BisectProgressFormatter.name, formatter_subclass)
+      end
+
+      it "prints the most minimal repro command it has found so far" do
```

Comment:
Again this could be useful documentation so I'd be ok with it being relocated to a scenario.